test/es6_proxy_strategy_test.dart

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) 2014, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+@TestOn('chrome')
+library polymer_interop.test.convert_test;
+
+import 'dart:html';
+import 'dart:js';
+import 'package:polymer_interop/polymer_interop.dart';
+import 'package:smoke/mirrors.dart' as smoke;
+import 'package:smoke/smoke.dart';
+import 'package:test/test.dart';
+import 'package:web_components/web_components.dart';
+
+main() async {
+  smoke.useMirrors();
+  await initWebComponents();
+
+  group('es6 proxy strategy', () {
+    setUpAll(() {
+      PolymerInteropConfiguration.listConversionStrategy = checkForEs6ProxySupport();
+      PolymerInteropConfiguration.mapConversionStrategy = checkForEs6ProxySupport();
+    });
+
+    tearDownAll(() {
+      PolymerInteropConfiguration.listConversionStrategy =
+          JsInteropStrategy.mixedMode;
+      PolymerInteropConfiguration.mapConversionStrategy =
+          JsInteropStrategy.mixedMode;
+    });
+
+    group('checks',() {
+      test('es6 proxy support',() {
+        expect(checkForEs6ProxySupport(),JsInteropStrategy.es6Proxy,reason:'es6 proxy not supported');
+      });
+    });
+
+    group('list', () {
+      test('push',() {
+        List myList = [];
+        JsArray jsArray = convertToJs(myList);
+        jsArray.callMethod('push',['a']);
+        expect(myList.length,1);
+        expect(myList[0],'a');
+      });
+
+      test('slice',() {
+        List myList = ['a','b','c'];
+        JsArray jsArray = convertToJs(myList);
+
+        JsArray sliced = jsArray.callMethod('slice',[1]);
+
+        expect(sliced.length,2);
+        expect(sliced[0],'b');
+        expect(sliced[1],'c');
+
+      });
+
+      test('delete',() {
+        List myList = ['a','b','c'];
+        JsArray jsArray = convertToJs(myList);
+
+        jsArray.callMethod('delete',['1']);
+
+        expect(jsArray.length,2);
+        expect(jsArray[0],'b');
+        expect(jsArray[1],'c');
+
+      },skip:'fix this test');
```

Comment:
I had to execute `Object.deleteProperty` or `delete array[index]`. There's no such a thing like `array.delete(i)`. I'll fix this.